### DISCUSSION TITLE
Incorrect Data Displayed for Binary Column When Querying SQL Server Table in DBeaver

### DISCUSSION THREAD

**Discussion by arsalanhub** [dbeaver/dbeaver#35720] [Q&A] [answered]
### Description

I am encountering an issue when querying a SQL Server table in DBeaver. Specifically, the query:

`SELECT MAX(__$start_lsn) FROM deleted.dbo.cdc_dbo_AideNote_CT_20240903`

returns the correct data when executed directly in SQL Server Management Studio (SSMS). However, when the same query is executed in DBeaver, it returns garbled or incorrect data.

![image](https://github.com/user-attachments/assets/39d2654d-3ef7-48ba-a7cc-bfd6bf284b1b)
![image](https://github.com/user-attachments/assets/fbb8f952-ba03-497d-8164-72654d1446d3)

### DBeaver Version

21.0.0

### Operating System

Windows

### Database and driver

Database: SQL Server
Class Name: com.microsoft.sqlserver.jdbc.SQLServerDriver
SQL Server Version: Microsoft SQL Server 2016 (SP3-OD) (KB5006943) - 13.0.6404.1 (X64)

### Steps to reproduce

Open DBeaver and connect to the SQL Server database.
Execute the query:

`SELECT MAX(__$start_lsn) FROM deleted.dbo.cdc_dbo_AideNote_CT_20240903`

Compare the results with the output from SQL Server Management Studio (SSMS). **You can also try this by adding any binary column and check as shown**.

![image](https://github.com/user-attachments/assets/c381af5f-16b3-4044-bfb7-e55fabb58d24)

**Expected Result**: The data returned by DBeaver should match the data returned by SSMS.

**Actual Result**: DBeaver displays incorrect or garbled data for the query.

**Comment by E1izabeth** [comment, marked as answer]
Hi, @arsalanhub 
You can switch Binary format view to Hex to get the desired behavior. It's in `context menu -> View/Format->Binary format`

![image](https://github.com/user-attachments/assets/37c5fc8d-1501-4ba4-8c86-64ed81a3e997)
Additionaly, it's possible to view cell values in the Hex editor in [Value viewer Panel](https://dbeaver.com/docs/dbeaver/Panels/#value-viewer).
![image](https://github.com/user-attachments/assets/e8b7256d-87ee-46d0-8dd4-285d1e44c663)

**Comment by ask9** [comment]
@E1izabeth Shouldn't it by default dispalyed in hex-format based on data type?
Because it doesn't always contain a string.

**Comment by E1izabeth** [comment]
We don't think hex should be default display format. The binary data may not be hex - in this case showing hex by default is not a good idea. It's not so rare strings are stored in the binary column.
